analysis/bulk-deconvolution/exploratory-notebooks/explore-epic-results.Rmd

```diff
@@ -0,0 +1,288 @@
+      "SCPCP000009" ~ "09: CNS tumors (TRef)", 
+      "SCPCP000017" ~ "17: Osteosarcoma (BRef)", 
+    ), 
+   # arrange in order of reference, not project
+   project = forcats::fct_relevel(project, "09: CNS tumors (TRef)", after = 2)
+  ) |>
+  dplyr::select(project, sample_id, epic_celltype, fraction)
+
+head(epic_df)
+```
+
+
+
+## Visualization
+
+### Proportions of cell types
+
+Considering all categories of cells, what proportions of cell types are represented across samples?
+
+
+```{r fig.height=8, fig.width=12}
+# Next, we'll define a vector of colors to use for plots which ensures that shared cell types have the same colors
+# hex codes are (mostly) from Okabe-Ito
+color_map <- c(
+  # shared cell types
+  "Bcells" = "#E69F00", 
+  "CD4_Tcells" = "#56B4E9", 
+  "CD8_Tcells" = "#009E73", 
+  "NKcells" =  "#F0E442",
+  # reference-specific cell types
+  "CAFs" = "#0072B2",
+  "Endothelial" = "#D55E00",
+  "Macrophages" = "#CC79A7",
+  "Monocytes" = "grey30", 
+  "Neutrophils" = "grey50",
+  "otherCells" = "grey95"
+)  
+celltype_levels <- c(names(color_map))
+  
+ggplot(epic_df) + 
+ aes(
+   x = sample_id, 
+   y = fraction, 
+   fill = forcats::fct_relevel(epic_celltype, celltype_levels)
+  ) +
+  geom_col() + 
+  facet_wrap(vars(project), scales = "free_x", nrow = 2) + 
+  scale_y_continuous(expand = c(0,0)) +
+  scale_fill_manual(values = color_map, name = "cell type") +
+  theme(
+    legend.position = "bottom", 
+    axis.text.x = element_blank()
+  )
+```
+
+In most samples, the vast majority of cells are `otherCells`, although there are two samples which are estimated to have ~100% cells classified; it is very likely that cells in these samples (and possibly more samples) are incorrectly classified.
+
+We also see these trends:
+
+* For the `TRef` projects:
+  * Macrophages are most commonly observed in `SCPCP000001` samples
+  * Endotheliel cells tend to dominate `SCPCP000001` and `SCPCP000002`
+* For the `BRef` projects:
+  * CD4 T-cells and monocytes are more prevalent in `SCPCP000017` compared to `SCPCP000006`
+  * CD8 T-cells dominate `SCPCP000006` cell types
+
+Let's have a quick look at the proportion of classified cells on its own:
+
+```{r}
+epic_classified_df <- epic_df |>
+  dplyr::filter(epic_celltype == "otherCells") |>
+  dplyr::mutate(fraction_classified = 1 - fraction) 
+
+ggplot(epic_classified_df) +
+  aes(x = project, y = fraction_classified) + 
+  geom_boxplot() + 
+  theme(axis.text.x = element_text(angle = 30, hjust = 1, size = rel(0.9)))
+
+```
+In the vast majority of samples, fewer than 25% of cells are classified, with the fewest cells classified in Wilms Tumor. 
+
+
+Notably, across samples, there appear to be very few B-cells inferred, and when they are inferred the frations are extremely small (to the point that we might essentially consider them to be 0).
+Let's look at this a bit more closely, zooming in on the section of the plot with most of the data:
+
+```{r, fig.width = 10}
```

Comment:
What's the reasoning behind zooming in on B cells here? I don't think you need it.